presto-hive/src/main/java/com/facebook/presto/hive/metastore/CachingHiveMetastore.java

```diff
@@ -549,6 +562,39 @@ public void alterPartition(String databaseName, String tableName, Partition part
         }
     }
 
+    @Override
+    public void createRole(String role, String grantor)
+    {
+        try {
+            delegate.createRole(role, grantor);
+        }
+        finally {
+            rolesCache.invalidateAll();
+        }
+    }
+
+    @Override
+    public void dropRole(String role)
+    {
+        try {
+            delegate.dropRole(role);
+        }
+        finally {
+            rolesCache.invalidateAll();
+        }
+    }
+
+    @Override
+    public Set<String> listRoles()
+    {
+        return get(rolesCache, "");
```

Comment:
`"unused"` instead, to make it clear it's irrelevant. Empty value looks like it's on purpose.
Or better make cache accepting `Void` instead of `String`.